azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -646,12 +646,19 @@ def compose_socket_options():
                 (socket.IPPROTO_TCP, socket.TCP_KEEPINTVL, INTERVAL_SECONDS),
                 (socket.IPPROTO_TCP, socket.TCP_KEEPCNT, MAX_FAILED_KEEPALIVES),
             ]
-        if sys.platform == "win32" and hasattr(socket, "SOL_SOCKET") and hasattr(socket, "SO_KEEPALIVE"):
-            options = [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)]
-            if hasattr(socket, "TCP_KEEPIDLE") and hasattr(socket, "TCP_KEEPCNT"):
-                options += [(socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, MAX_IDLE_SECONDS), (socket.IPPROTO_TCP, socket.TCP_KEEPCNT, MAX_FAILED_KEEPALIVES)]
-            return options
-        if sys.platform == "darwin" and hasattr(socket, "SOL_SOCKET") and hasattr(socket, "SO_KEEPALIVE") and hasattr(socket, "IPPROTO_TCP"):
+        elif (

```

Comment:
I think you'll conflict here with @yihezkel

Reply:
I merged from master after his changes, didn't I?
maybe I didn't push it

Reply:
This makes more sense.
Feel free to ignore